pkg/plugins/resources/k8s/native/api/v1alpha1/mesh_gateway_config.go

```diff
@@ -86,6 +105,49 @@ type MeshGatewayServiceSpec struct {
 
 // +k8s:deepcopy-gen=true
 
+// MeshGatewayPodSpec holds customizable fields of a Service spec.
+type MeshGatewayPodSpec struct {
+	// ServiceAccountName corresponds to PodSpec.ServiceAccountName.
+	// +optional
+	ServiceAccountName string `json:"serviceAccountName,omitempty"`
+
+	// PodSecurityContext corresponds to PodSpec.SecurityContext
+	// +optional
+	PodSecurityContext `json:"securityContext,omitempty"`
+
+	// Container corresponds to PodSpec.Container
+	// +optional
+	Container `json:"container,omitempty"`
```

Comment:
Are these all supposed to be embedded?

```suggestion
	Container Container `json:"container,omitempty"`
```